File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    return (count * (count+1)/2)**2
    # return sum(range(count + 1))**2


def sum_of_squares(count):
    return (count * (count+1) * (2*count+1)) / 6
    # return sum(x**2 for x in range(count + 1))


def difference_of_squares(count):
    return abs(sum_of_squares(count) - square_of_sum(count))

File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'range.dart';

typedef IntFn = int Function(int x);

class DifferenceOfSquares {
  int _ident(int n) => n;
  int _square(int n) => n * n;

  int squareOfSum(int n) => _doSum(n, _ident, _square);
  int sumOfSquares(int n) => _doSum(n, _square, _ident);

  int _doSum(int n, IntFn innerFn, IntFn outerFn) {
    int sum = 1.upto(n).fold(0, (s, i) => s + innerFn(i));
    return outerFn(sum);
  }

  int differenceOfSquares(int n) => (sumOfSquares(n) - squareOfSum(n)).abs();
}
